cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -101,6 +102,26 @@ func (p *BicepProvider) GetDeployment(ctx context.Context, scope infra.Scope) *a
 		})
 }
 
+// convert from: sdk client outputs: interface{} to map[string]azcli.AzCliDeploymentOutput
+// sdk client parses http response from network as an interface{}
+// this function keeps the compatibility with the previous AzCliDeploymentOutput model
+func createDeploymentOutput(rawOutputs interface{}) (result map[string]azcli.AzCliDeploymentOutput) {
+	if rawOutputs == nil {
+		return make(map[string]azcli.AzCliDeploymentOutput, 0)
+	}
+
+	castInput := rawOutputs.(map[string]interface{})
```

Comment:
Should we move this is our azure client layer? The weak type `interface{}` doesn't seem ideal for most usage. Do you also know the rationale behind the SDK? Is outputs supposed to be dynamic?

Reply:
This is how the Service defines the swagger :(, and then the sdk client is auto generated from the spec.
Swagger does not have a type for outputs, other than `object`, see: https://learn.microsoft.com/en-us/rest/api/resources/deployments/get#deploymentpropertiesextended